packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -158,8 +191,21 @@ class ExpansionPanelList extends StatelessWidget {
               isExpanded: _isChildExpanded(index),
               padding: const EdgeInsets.all(16.0),
               onPressed: (bool isExpanded) {
-                if (expansionCallback != null)
-                  expansionCallback(index, isExpanded);
+                if (widget.expansionCallback != null)
+                  widget.expansionCallback(index, isExpanded);
+
+                setState(() {
+                  if(!widget.allowMultiplePanelsOpen) {
+                    for (int cIndex = 0; cIndex < widget.children.length; cIndex++) {
```

Comment:
nit: don't use abbreviations in variable names
nit: use `+= 1` rather than `++`